### DISCUSSION TITLE
Is there any method to remove or ignore the footer and header text?

### DISCUSSION THREAD

**Discussion by hemilparmar** [py-pdf/pypdf#1354] [Q&A]
can we remove header and footer from pdf while extracting the text?

**Comment by pubpub-zz** [comment]
a new feature is under publication 
https://github.com/py-pdf/PyPDF2/pull/1252 thanks to  @srogmann
just a little of patience ...

**Reply by hemilparmar** [reply]
I have tried using visitor function but that does not work for all pdfs. we have to change the value of y0, y1 according to pdf structure.
i have tried to convert first pdf to doc and extract the header and footer from doc but it is also not working properly. so can we do something that parser will extract the footer and header automatically ?

**Reply by MartinThoma** [reply]
There is no way to do this reliably for all kinds of pdfs. With machine leaning or some heuristics you can get a long way. For example, looking for repeated elements in the beginning /end of each page (potentially differentiating even/odd pages, considering title / chapter/filler pages). However, that is not something PyPDF2 itself should do
